### ISSUE TITLE
support for pathlib.Path objects

### ISSUE THREAD

**Issue by gdementen** [Roche/pyreadstat#110]
**Describe the issue**

It would be convenient if pyreadstat supported pathlib.Path objects natively.

**To Reproduce**

```python
>>> from pathlib import Path
>>> src_path = Path('test.sas7bdat')
>>> df, meta = pyreadstat.read_sas7bdat(src_path)
[...]
TypeError: Argument 'filename_path' has incorrect type (expected str, got WindowsPath)
```

**File example**

N/A

**Expected behavior**

Work 😉. That is without using the obvious workaround:
```python
>>> df, meta = pyreadstat.read_sas7bdat(str(src_path))
```

**Setup Information:**

pyreadstat 1.0.8 on Anaconda Python3.8 on Windows (installed using conda) but I think this is all irrelevant

**Comment by ofajardo** [comment]
thanks for the suggestion. I internally have to pass a string to the C function, but I guess the only thing I have to do internally is str(src_path) if the type of the object is Pathlib? OR are there other aspects pitfalls to take into account?

**Comment by gdementen** [comment]
> I guess the only thing I have to do internally is str(src_path) if the type of the object is Pathlib? OR are there other aspects pitfalls/corner cases to take into account?

There is no pitfall that I know of. I have always used str(). I **think** str() is the canonical way to get a "raw" filesystem path but you **might** need to use bytes() on unix as per https://docs.python.org/3/library/pathlib.html#operators but I don't really know.

**Comment by ofajardo** [comment]
fixed in dev, will become available on the next release

**Comment by gdementen** [comment]
Wow! Thanks for the superb support (as always)!

**Comment by ofajardo** [comment]
fixed in version 1.0.9. Let me know if something is not working as expected.
